app/models/job_execution.rb

```diff
@@ -57,13 +60,13 @@ def close
   end
 
   def stop!
-    if @thread
-      @executor.stop! 'INT'
-      return if @thread.join(stop_timeout)
+    return unless @thread&.alive?
```

Comment:
stopping when it is not running should never happen ... could also raise here ... basically trying to avoid that a failed/succeeded job gets changed to cancelled